lnd.go

```diff
@@ -59,24 +59,34 @@ var (
 	networkDir string
 )
 
+// ListnerWithSignal is a net.Listner that has an additional Ready channel that
+// will be closed when a server starts listening.
+type ListenerWithSignal struct {
+	net.Listener
+
+	// Ready will be closed by the server listening on Listener.
+	Ready chan struct{}
```

Comment:
Could we make this `<-chan struct{}` to make sure we only ever read from it?

Reply:
We can't really, since we are reading from it one place, and closing it the other place.